src/perform_step/bdf_perform_step.jl

```diff
@@ -936,21 +936,38 @@ function perform_step!(integrator,cache::QNDFCache,repeat_step=false)
     end
 
     if k > 1
-      @.. utilde = (κ*γₖ[k-1] + inv(k)) * D[k]
+      @.. utilde = (integrator.alg.kappa[k-1]*γₖ[k-1] + inv(k)) * D[k]
       calculate_residuals!(atmp, utilde, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm, t)
       cache.EEst1 = integrator.opts.internalnorm(atmp,t)
+    else
+      cache.EEst1 = Inf
     end
     backward_diff!(cache,D,D2,k+1,false)
     @.. tmp = u - uprev
     for i = 1:(k+1)
       @. tmp -= D2[i,1]
     end
-    @.. utilde = (κ*γₖ[k+1] + inv(k+2)) * tmp
+    if k < 5
+    @.. utilde = (integrator.alg.kappa[k+1]*γₖ[k+1] + inv(k+2)) * tmp
     calculate_residuals!(atmp, utilde, uprev, u, integrator.opts.abstol, integrator.opts.reltol, integrator.opts.internalnorm, t)
     cache.EEst2 = integrator.opts.internalnorm(atmp,t)
+    else
+      cache.EEst2 = Inf
+    end
     # cnt == 1
   end # integrator.opts.adaptive
 
+  if integrator.EEst > one(integrator.EEst)
+    for i = 1:5
+      fill!(D[i], zero(eltype(u)))
+    end
+    for i = 1:6, j = 1:6
```

Comment:
there must be a better way to fill it recursively with zeros

Reply:
I don't know of one 🥘

Reply:
Doesn't solve the problem, but one could also just iterate over `D` and `D2` without writing down the indices explicitly. Would still use `fill!` recursively but one doesn't have to know and write down the dimensions explicitly.